src/pybind/mgr/dashboard/rgw.py

```diff
@@ -0,0 +1,45 @@
+# -*- code: utf-8 -*-
+
+import json
+import re
+import rados
+from remote_view_cache import RemoteViewCache
+
+import logging
+
+log = logging.getLogger("dashboard")
+
+class RGWDaemons(RemoteViewCache):
+
+    def _get(self):
+        daemons = self.get_daemons()
+        return {
+            'daemons': daemons,
+        }
+
+    def get_daemons(self):
+        daemons = []
+        for server in self._module.list_servers():
+            for service in server['services']:
+                if service['type'] == 'rgw':
+                    metadata = self._module.get_metadata('rgw', service['id'])
+                    status = self._module.get_daemon_status('rgw', service['id'])
+                    try:
+                        status = json.loads(status['json'])
+                    except:
+                        status = {}
+                    
+                    # extract per-daemon service data and health
+                    daemon = {
+                        'id': service['id'],
+                        'version': metadata['ceph_version'],
+                        'server_hostname': server['hostname'],
+                        'service': service,
+                        'server': server,
+                        'metadata': metadata,
+                        'status': status,
+                        'url': "/rgw/detail/{0}".format(service['id'])
```

Comment:
I suggest leaving out the /detail/ part, and just use /rgw for the list and `/rgw/<id>` for the detail

Reply:
sorry, John,  how could I get this `<id>` in `RGWEndpoint::index` function. I tested with `index(self, rgw_id=None)` and `index(self, rgw_id)`, cherrypy could not dispatch request correctly as expected.  we use `/osd/perf/<id>` in OSDEndpoint, not `/osd/<id>`.  I wonder whether cherrypy could support this url pattern. I 'm looking cherrypy's document.

Reply:
It's a little awkward, but I think in this case you can prefix a `@cherrypy.popargs('rgw_id')` to the `RGWEndpoint` class (http://docs.cherrypy.org/en/latest/advanced.html#restful-style-dispatching), and then the rgw_id argument to `index` will work.